### DISCUSSION TITLE
Adding rename symbol to a language extension

### DISCUSSION THREAD

**Discussion by gesslar** [microsoft/vscode-discussions#593] [Extension Development QnA] [answered]
Hi there, I'm a collaborator on a language extension (LPC). We have snippets and are in the middle of adding more options to our syntax highlighting. But, I was wondering, how would we go about adding support for rename symbol. Pressing F2 doesn't do anything, but I am accustomed to it when I write JavaScript.

My GoogleFu has been failing me on trying to find this information. Would appreciate any pointers in a helpful direction. Thank you! ♥

**Comment by r3m0t** [comment, marked as answer]
Either https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_rename or https://code.visualstudio.com/api/references/vscode-api#RenameProvider depending on whether you're writing an LSP or pure VS Code extension

**Reply by gesslar** [reply]
Thank you for your response. Can you think of an example where I might draw from? We are not employing a language server, so it looks like the second would be the option would be the one to examine.
